### ISSUE TITLE
[BUG] Can't play anything. 

### ISSUE THREAD

**Issue by Arathonk** [nixgates/plugin.video.seren#949]
**Describe the bug**
I can't play any sources, gives a general Seren error/no playable sources found error. "error code 37" RD is active and re authorised. 

**To Reproduce**
Play any episode/file

**Expected behavior**
Play the episode/file

**Screenshots**
-

**Kodi Version (please complete the following information):**
 - Kodi Version 20.3

**Additional context**
Kodi logfile: https://paste.kodi.tv/aluhasasik
Seren version latest 3.0.1

**Comment by Arathonk** [comment]
Apperently it's the RD API because of some infringement issues?

**Comment by miminno** [comment]
Same issue here. Any workarounds?

**Comment by henryjfry** [comment]
If you are adventurous you could try my attempt here:

https://gist.github.com/henryjfry/725cce9ce0222b063e5e40a7c571d032

Appears to work.

You'd probably need to extend your scraping timeout

**Comment by Arathonk** [comment]
> If you are adventurous you could try my attempt here:
> 
> https://gist.github.com/henryjfry/725cce9ce0222b063e5e40a7c571d032
> 
> Appears to work.
> 
> You'd probably need to extend your scraping timeout

What do we need to do with that?

**Comment by henryjfry** [comment]
Replace the contents of:

~/.kodi/addons/plugin.video.seren/resources/lib/debrid/real_debrid.py

**Comment by peno64** [comment]
Don't you guys know that France federal government has instructucted real debrid to stop providing illegal video content? They had to disable some API functions and that is the reason why all addons, not only Seren, doesn't work anymore as it worked. RD also says that they will have to remove all illegal content from their servers.

**Comment by Arathonk** [comment]
Yes, I found out after opening the issue. But there seems to be solutions, like the one @henryjfry posted, I'll try that later. It also works again on Torrentio@Stremio

**Comment by henryjfry** [comment]
> Don't you guys know that France federal government has instructucted real debrid to stop providing illegal video content? They had to disable some API functions and that is the reason why all addons, not only Seren, doesn't work anymore as it worked. RD also says that they will have to remove all illegal content from their servers.

They disabled the "instantAvailability" api endpoint.
But unless they effectively stop providing any kind of torrent functionality via API thats all thats has changed so far.
They may have stepped up DMCA action but until that time they wont be removing anything unless explicitly asked. Because if they start removing stuff without explicit instruction they stand to lose money.

As is its basically only useful for many people with the torrents functionality. Without that they cant possibly survive as a viable service in my opinion. So stuff will start going, and admittedly a  court order to terminate a specific API endpoint is very worrying.
But its definetly not the end. Not yet.

**Comment by peno64** [comment]
@henryjfry 

Could it be that your change to real_debrid.py is based on an older version of that file? I see differences that don't have a purpose to work-around this problem.

**Comment by henryjfry** [comment]
> @henryjfry
> 
> Could it be that your change to real_debrid.py is based on an older version of that file? I see differences that don't have a purpose to work-around this problem.

Yeah probably, i figured it wouldn't have changed too much but it is from v3.0.0.0

But to nip any future queries in the bud if a direct copy replacement doesnt work, wait for official addon updates.

**Comment by peno64** [comment]
@henryjfry 

I will try your change in a moment. Can you elaborate on your statement "You'd probably need to extend your scraping timeout".
Where do you make this change and what value do you use?

**Comment by henryjfry** [comment]
In seren settings under scraping i think

**Comment by peno64** [comment]
Just tried the proposed change from @henryjfry and it works. Even without changing the timeout. 

Maybe a zip of this should be made such that other people can also install this.

**Comment by peno64** [comment]
@henryjfry 

Looking at the change you did in that file I must say you are a magician. Working around the instantAvailability api call and having made that fix so fast. Wow.

**Comment by Arathonk** [comment]
I checked the fix aswell, and it does work! However it's a bit slow, takes a little while to find sources, thanks bud!

**Comment by henryjfry** [comment]
Yeah there is likely some kind of optimization which can be done, a duplicate/recent cache lookup check which would potentially eliminate some unnecessary calls.
Additionally filtering out files at the select_all stage based on the source filtering criteria could save some calls.

I'm fairly certain that what now happens is it adds a torrent, queries if it's downloaded, deletes it. Then queries all the cached torrents by adding them again to determine which one to stream.

But definitely some doubling of the lookups needed is happening.
However this was the quickest way to get the current code working so 🤷‍♂️

**Comment by Arathonk** [comment]
Yes that's what I figured, but at least it's working, saved a lot of headaches.

**Comment by Mike580** [comment]
Sorry, could you please elaborate on how to replace the contents of ~/.kodi/addons/plugin.video.seren/resources/lib/debrid/real_debrid.py?  Thanks in advance

**Comment by TwilightMercy** [comment]
> If you are adventurous you could try my attempt here:
> 
> https://gist.github.com/henryjfry/725cce9ce0222b063e5e40a7c571d032
> 
> Appears to work.
> 
> You'd probably need to extend your scraping timeout

How the workaround works?

**Comment by Arathonk** [comment]
You have to go to the file location(just look up online how to get there),
~/.kodi/addons/plugin.video.seren/resources/lib/debrid/real_debrid.py 
And then replace the contents or replace the entire file with this:
https://gist.github.com/henryjfry/725cce9ce0222b063e5e40a7c571d032
You can either copy and paste the text, or you can download the file as a zip. 
It's really easy.

**Comment by shlomy79** [comment]
> You have to go to the file location(just look up online how to get there), ~/.kodi/addons/plugin.video.seren/resources/lib/debrid/real_debrid.py And then replace the contents or replace the entire file with this: https://gist.github.com/henryjfry/725cce9ce0222b063e5e40a7c571d032 You can either copy and paste the text, or you can download the file as a zip. It's really easy.

not working  i copied thr file over the old one then when i opend the kodi its give error on twilight addon i tring and getting to load a tv show no links

**Comment by henryjfry** [comment]
> > You have to go to the file location(just look up online how to get there), ~/.kodi/addons/plugin.video.seren/resources/lib/debrid/real_debrid.py And then replace the contents or replace the entire file with this: https://gist.github.com/henryjfry/725cce9ce0222b063e5e40a7c571d032 You can either copy and paste the text, or you can download the file as a zip. It's really easy.
> 
> not working i copied thr file over the old one then when i opend the kodi its give error on twilight addon i tring and getting to load a tv show no links

Then i suspect you didnt do it correctly.
But without any error messages for context thats all I can say.

Dont know what twilight is or has to do with seren

**Comment by Jbone97** [comment]
> Replace the contents of:
> 
> ~/.kodi/addons/plugin.video.seren/resources/lib/debrid/real_debrid.py

I can’t seem to get to this file location. So here I am begging for help. 🤦‍♂️

**Comment by piergni** [comment]
> > Replace the contents of:
> > ~/.kodi/addons/plugin.video.seren/resources/lib/debrid/real_debrid.py
> 
> I can’t seem to get to this file location. So here I am begging for help. 🤦‍♂️

locations:
https://kodi.wiki/view/Kodi_data_folder

depends on the device you're trying to do this on....  on the newest version of android, you can't access Android/data/ ( this whole folder is empty/locked down ) so it's not possible.  I spent hours yesterday trying to figure this out just to give up and switch to a different add-on. Was trying to do this on the newest model Firestick.

**Comment by fade2gray** [comment]
@henryjfry's method worked for me on an RPi5 👍, and I will be trying it on an older Fire Stick using this guide ...
https://www.youtube.com/watch?v=EAn1gatsTfA

**Comment by peno64** [comment]
@piergni 

It is perfectly possible on an android, even on the newest. However you must use the kodi file manager for that. Put the file on a shred location like a network drive or an usb stick and then you can copy the file via kodi's file manager from that shared location to the addon folder.

**Comment by piergni** [comment]
interesting @peno64 

if I hadn't found a work around I'd try that, but swapping add-ons for me isn't that big a deal

**Comment by fade2gray** [comment]
Re my previous comment: https://github.com/nixgates/plugin.video.seren/issues/949#issuecomment-2495993358

The Xplore app used in the video is no longer foc to use Wi-Fi or Ftp file transfer, but you can use Ftp file transfer foc with the ES File Explorer app.

**Comment by liquidcms** [comment]
Some tips for those with Nvidia Shield to use Henry's fix:
- on Shield: Settings - Device Preferences - Storage - Transfer files over local network (even if enabled, click it to see info/options)
- note ip/username/password (and make sure enabled, obviously)
- got to pc on same network (Windows) and in file explorer enter \\[ip address]
- it will ask for username/password
- brings you to \internal
- go to: \internal\Android\data\org.xbmc.kodi\files\.kodi\addons\plugin.video.seren\resources\lib\debrid and replace real-debrid.py with Henry's version.

I did this while Kodi was stopped; but as we know stopping an app in Shield doesnt really stop it so i had to go to Settings - Apps - Kodi and Force stop and then restart Kodi for change to take affect.

Thanks Henry for the fix.

**Comment by djmdesign** [comment]
Works, its not perfect, takes a long time for a movie torrent to start, but its better than nothing.
Tv shows are quick, and im using a 1gig down 750 up fiber connection
just replace the .py file in your appdata roaming kodi plugins - seren addon. search for it in appdtat roaming.
 THANK YOU VERY MUCH!!!!

**Comment by elmedico16912** [comment]
> Some tips for those with Nvidia Shield to use Henry's fix:
> 
>     * on Shield: Settings - Device Preferences - Storage - Transfer files over local network (even if enabled, click it to see info/options)
> 
>     * note ip/username/password (and make sure enabled, obviously)
> 
>     * got to pc on same network (Windows) and in file explorer enter \[ip address]
> 
>     * it will ask for username/password
> 
>     * brings you to \internal
> 
>     * go to: \internal\Android\data\org.xbmc.kodi\files.kodi\addons\plugin.video.seren\resources\lib\debrid and replace real-debrid.py with Henry's version.
> 
> 
> I did this while Kodi was stopped; but as we know stopping an app in Shield doesnt really stop it so i had to go to Settings - Apps - Kodi and Force stop and then restart Kodi for change to take affect.
> 
> Thanks Henry for the fix.

hello liquidcms. i have an Nvidia Shield. when i network my PC and go to location \internal\Android\data\org.xbmc.kodi\files i don't see anything in the 'files' folder. Seren has been properly installed and was working fine prior to this issue. Any further help regarding fixing this issue on an Nvidia Shield? Thanks

**Comment by elmedico16912** [comment]
disregard, i had to enable show hidden files on the PC

**Comment by sgh2415** [comment]
Hey it works but it takes a few minutes even after the scrapeout is extended. Is there anyway to resolve that or just how it will be? Thank you, either way!

**Comment by henryjfry** [comment]
I've deleted the public gist because I didn't want it widely publicised.

I'm worried about mass adoption and potential harm to RDs servers

Folk on Reddit can be your source for help now I was my hands of it.

**Comment by peno64** [comment]
Well widely published is not really an argument since stremio also adapted its code to work and in kodi Fen Light is also working now so people are fully working again with RD. Wonder how long it will take before another api will be dropped.

**Comment by henryjfry** [comment]
Yes but I know how I adapted this code and there isnt any rate limiting or caching or anything like that.  It literally hammers the RD API with addmagnet and delete calls.
Possibly stremio/fen light have implemented something better? Possibly not. But i know that my code is probably not advisable to be used by everyone in the long term.

I dont want to be partly responsible for accelerating RD's demise

**Comment by henryjfry** [comment]
This version is better and only check the RD API when you try to play a file:

https://raw.githubusercontent.com/nixgates/plugin.video.seren/f01044749fe7468df1f9c4588f6b664017bd4194/resources/lib/debrid/real_debrid.py

Scraping is much, much faster.

**Comment by peno64** [comment]
@henryjfry 
I don't see much difference with the previous version. In fact the only differences are that you removed the commented code check_hash and _check_hash_thread and in _check_hash_thread the variable hash_string is removed.

**Comment by bbviking** [comment]
> This version is better and only check the RD API when you try to play a file:
> 
> https://raw.githubusercontent.com/nixgates/plugin.video.seren/f01044749fe7468df1f9c4588f6b664017bd4194/resources/lib/debrid/real_debrid.py
> 
> Scraping is much, much faster.

The fix also requires that this getSources.py replace the existing one in resources/lib/modules

https://gist.githubusercontent.com/bbviking/22d77506ea9ec28e424b072e2ac46bec/raw/c20f693f960d222fc2d99dbe63e2b25034c9a346/getSources.py

The real_debrid.py from my pull request is functionally the same as yours, but in the _realdebrid_worker function I skipped the hash checking.

I might post a zipped up version of seren with the fix on reddit in a bit so hopefully people will stop hammering RD's API lol

**Comment by bbviking** [comment]
> @henryjfry I don't see much difference with the previous version. In fact the only differences are that you removed the commented code check_hash and _check_hash_thread and in _check_hash_thread the variable hash_string is removed.

Does replacing the getSources.py file improve things for you?

**Comment by peno64** [comment]
> Does replacing the getSources.py file improve things for you?

@bbviking 

yes it does. 
I see that your PR contains even more changes.
Will clone your branch and make a zip from it and install to test it

**Comment by peno64** [comment]
@bbviking 

Made a zip from your version 3.0.5 and installed it. Seems to work fine and indeed alot faster in scraping. Well done. Haven't tried your added options though.

**Comment by Arathonk** [comment]
Can you share that zip so we can test it aswell?

**Comment by peno64** [comment]
Surf to https://github.com/bbviking/plugin.video.seren
Press the green Button "Code"
Choose Download ZIP
unzip that file to a new location
rename plugin.video.seren-master to plugin.video.seren
zip that again to plugin.video.seren.zip (note that the folder plugin.video.seren must be at the root of the zip)
That zip file can now be used in kodi via install from zip
so in fact you new created zip is the same as the downloaded one with the same structure except that plugin.video.seren-master is renamed to plugin.video.seren
If you use 7-zip you can do that even without unzipping/zipping the downloaded zip

**Comment by rpartapsing** [comment]
Nice it works!! For me adding the zip through UI failed. Copying the plugin.video.seren folder into Roaming\Kodi\addons (after removing the existing folder) did the trick

**Comment by bbviking** [comment]
> @bbviking
> 
> Made a zip from your version 3.0.5 and installed it. Seems to work fine and indeed alot faster in scraping. Well done. Haven't tried your added options though.

Thanks! Glad it worked for you

**Comment by bbviking** [comment]
> Nice it works!! For me adding the zip through UI failed. Copying the plugin.video.seren folder into Roaming\Kodi\addons (after removing the existing folder) did the trick

I think it might have something to do with where the zip you're trying to install is stored. It failed when I tried to install the zip when it was located in my downloads folder, but desktop was fine.

**Comment by arturochent** [comment]
Does this also works for PC?

**Comment by bbviking** [comment]
> Does this also works for PC?

It works on PC yeah

**Comment by TroglodyteGuy** [comment]
Same issue, cannot play anything on my Onn Android device (either of two). Both have the same symptoms -- Playback failed. Seren finds plenty of "Total" and "Torrents" files, but ZERO "Playable" returned. Did not try on PC as I do not use Seren on the PC.

**Comment by Jbone97** [comment]
It took me a while when I was able to sit down and apply the fix, I’m persistent, but by no means a computer wiz.  Many thanks to all your combined wizardry 🧙!  FYI zip didn’t work for me in either storage location. And I have no idea where that “roaming folder” is hiding lol

**Comment by bills176** [comment]
My setup is on a Win 11 machine. I also have Premiumize connected. I installed the RD fix here and it seems to work. However when I now scrape a TV show I only get RD response and no returns from Premiumize. Scraping movies works as expected. Rolling back Seren returns Premiumize. Seems something else has changed. I appreciate all the help of the folks here.

**Comment by bbviking** [comment]
> My setup is on a Win 11 machine. I also have Premiumize connected. I installed the RD fix here and it seems to work. However when I now scrape a TV show I only get RD response and no returns from Premiumize. Scraping movies works as expected. Rolling back Seren returns Premiumize. Seems something else has changed. I appreciate all the help of the folks here.

Interesting, when you disable real debrid do the premiumize results come back?

And are you sure there aren't any premiumize results? Perhaps they're at the bottom depending on how seren is prioritising sources

**Comment by DeAngryLeprejuan** [comment]
I find, If you turn on Manual selection for sources, You can select the Torrent, And it will load it to Read Debrid.

Real Debrid Will no long give streams Services like Kodi and Streamio "Instant Avalibility" Looks like I'll be moving over to Preimuimize.me

**Comment by bills176** [comment]
> > My setup is on a Win 11 machine. I also have Premiumize connected. I installed the RD fix here and it seems to work. However when I now scrape a TV show I only get RD response and no returns from Premiumize. Scraping movies works as expected. Rolling back Seren returns Premiumize. Seems something else has changed. I appreciate all the help of the folks here.
> 
> Interesting, when you disable real debrid do the premiumize results come back?
> 
> And are you sure there aren't any premiumize results? Perhaps they're at the bottom depending on how seren is prioritising sources

There are no premiumize results. I cleared the RD authorization and I still get RD results and no premiumize results in TV shows. Get both in movies with RD disabled. When I re-enable RD my account info shows. I reset to defaults for that settings page and it cleared my account data. I reauthorized premiumize, left RD blank and still get RD results in TV and Movies. I may have to roll back and use only premiumize. Has anyone else tried revoking RD authorization?

**Comment by bbviking** [comment]
> > > My setup is on a Win 11 machine. I also have Premiumize connected. I installed the RD fix here and it seems to work. However when I now scrape a TV show I only get RD response and no returns from Premiumize. Scraping movies works as expected. Rolling back Seren returns Premiumize. Seems something else has changed. I appreciate all the help of the folks here.
> > 
> > 
> > Interesting, when you disable real debrid do the premiumize results come back?
> > And are you sure there aren't any premiumize results? Perhaps they're at the bottom depending on how seren is prioritising sources
> 
> There are no premiumize results. I cleared the RD authorization and I still get RD results and no premiumize results in TV shows. Get both in movies with RD disabled. When I re-enable RD my account info shows. I reset to defaults for that settings page and it cleared my account data. I reauthorized premiumize, left RD blank and still get RD results in TV and Movies. I may have to roll back and use only premiumize. Has anyone else tried revoking RD authorization?

For premiumize results not showing, have you tried disabling preemptive termination? I think seren might not be including those results because scraping has already been terminated. I have tested it myself and with that disabled, I was able to get premiumize, real debrid, all debrid results to show together

Rescrape just to double check. Rescraping should also fix the issue you were having with RD results showing up, seren still has them cached from when you had it enabled.

**Comment by bills176** [comment]
> 
> For premiumize results not showing, have you tried disabling preemptive termination? I think seren might not be including those results because scraping has already been terminated. I have tested it myself and with that disabled, I was able to get premiumize, real debrid, all debrid results to show together
> 
> Rescrape just to double check. Rescraping should also fix the issue you were having with RD results showing up, seren still has them cached from when you had it enabled.

You were right. Good catch. Prior to disabling preemptive termination I did try clearing cache and rescraping. It was acting goofy. Anyways seems okay now. Thanks for the advice and help on all this.

**Comment by Jbone97** [comment]
> @henryjfry's method worked for me on an RPi5 👍, and I will be trying it on an older Fire Stick using this guide ... https://www.youtube.com/watch?v=EAn1gatsTfA

Were you successful in doing the fix on a fire stick? I was able to get it done on one of my shields going to do my 2nd one and then attempt the firestick…

**Comment by bbviking** [comment]
> You were right. Good catch. Prior to disabling preemptive termination I did try clearing cache and rescraping. It was acting goofy. Anyways seems okay now. Thanks for the advice and help on all this.

No worries, I'm just glad I didn't create any new bugs haha. I felt the same way when I tested it, it did seem to behave weirdly initially but seemed fine soon enough. Maybe I didn't properly clear the cache or something.

**Comment by bbviking** [comment]
> Were you successful in doing the fix on a fire stick? I was able to get it done on one of my shields going to do my 2nd one and then attempt the firestick…

I've posted a reddit thread about the update that comes with installation instructions [LINK](https://www.reddit.com/r/Addons4Kodi/comments/1h0msfp/seren_306_all_debrid_fix_added/)

**Install Instructions**


1. To install this update, first have a previous version of seren installed to ensure that you have the required dependencies.

2. Next in Kodi go to 'Settings' -> 'File Manager' -> 'Add Source'

3. For the source enter https://bbviking.github.io/ and name it anything you want (probably bbviking for simplicity)

4. You will need to install [plugin.video.seren-3.0.6.zip](http://plugin.video.seren-3.0.6.zip/) by going to 'Add-ons' -> 'Install from Zip' , choosing the source you just added and then the zip file. If you don't know how to do that, you can follow this guide: https://kodi.wiki/view/Add-on_manager#How_to_install_from_a_ZIP_file

**Comment by NorthEndSauce** [comment]
> > Were you successful in doing the fix on a fire stick? I was able to get it done on one of my shields going to do my 2nd one and then attempt the firestick…
> 
> I've posted a reddit thread about the update that comes with installation instructions [LINK](https://www.reddit.com/r/Addons4Kodi/comments/1h0msfp/seren_306_all_debrid_fix_added/)
> 
> **Install Instructions**
> 
> 1. To install this update, first have a previous version of seren installed to ensure that you have the required dependencies.
> 2. Next in Kodi go to 'Settings' -> 'File Manager' -> 'Add Source'
> 3. For the source enter https://bbviking.github.io/ and name it anything you want (probably bbviking for simplicity)
> 4. You will need to install [plugin.video.seren-3.0.6.zip](http://plugin.video.seren-3.0.6.zip/) by going to 'Add-ons' -> 'Install from Zip' , choosing the source you just added and then the zip file. If you don't know how to do that, you can follow this guide: https://kodi.wiki/view/Add-on_manager#How_to_install_from_a_ZIP_file


thanks so much for this! I’m not the greatest online anymore or with Firesticks but created an account just to thank you! Got my shows back

**Comment by fade2gray** [comment]
@Jbone97 Yes, I followed the guide to install 'ES File Explorer' instead of 'Xplore'.

In ESFE, go to Display Settings and enable 'Display Hidden Files', then find and select 'View on PC' followed by 'Turn On' - a ftp address will be displayed which can be entered in Windows Explorer or other ftp client.

**Comment by arturochent** [comment]
> Surf to https://github.com/bbviking/plugin.video.seren Press the green Button "Code" Choose Download ZIP unzip that file to a new location rename plugin.video.seren-master to plugin.video.seren zip that again to plugin.video.seren.zip (note that the folder plugin.video.seren must be at the root of the zip) That zip file can now be used in kodi via install from zip so in fact you new created zip is the same as the downloaded one with the same structure except that plugin.video.seren-master is renamed to plugin.video.seren If you use 7-zip you can do that even without unzipping/zipping the downloaded zip

Wonderful!! Kudos on this one!! Install was flawless

**Comment by liquidcms** [comment]
I have 3.0.5 installed. Add new source to bbviking's repo. When I add from zip and go to this source I see zip for 3.0.61. Selecting this does nothing except remove my previous version.

If I reboot kodi, my old version appears again.

**Comment by athefirestick** [comment]
I can pull several resources, but I only end up with one cloud resource. The cloud resource appears to download, but I can't access the location to play the file. Where would this resource be stored, and how do I play the file? I did access Trakt directly and noticed that some TV/Movies were blocked from view and required a subscription. If there is a fix, then let's have it. Nobody wants to take out 10 subscriptions to view movies and tv shows. What happened to pay-for-view?

**Comment by bbviking** [comment]
> thanks so much for this! I’m not the greatest online anymore or with Firesticks but created an account just to thank you! Got my shows back

No worries! I'm glad it worked for you :)

**Comment by bbviking** [comment]
> **Install Instructions**
> 
> 1. To install this update, first have a previous version of seren installed to ensure that you have the required dependencies.
> 2. Next in Kodi go to 'Settings' -> 'File Manager' -> 'Add Source'
> 3. For the source enter https://bbviking.github.io/ and name it anything you want (probably bbviking for simplicity)
> 4. You will need to install [plugin.video.seren-3.0.6.zip](http://plugin.video.seren-3.0.6.zip/) by going to 'Add-ons' -> 'Install from Zip' , choosing the source you just added and then the zip file. If you don't know how to do that, you can follow this guide: https://kodi.wiki/view/Add-on_manager#How_to_install_from_a_ZIP_file

I have updated the RD workaround implementation that should fix some issues that people were having. The same installation instructions I posted still apply.

You can find the reddit post I made for this update here:

https://www.reddit.com/r/Addons4Kodi/comments/1h3j6c2/seren_3062_improved_rd_fix_implementation/

**Comment by bbviking** [comment]
> I have 3.0.5 installed. Add new source to bbviking's repo. When I add from zip and go to this source I see zip for 3.0.61. Selecting this does nothing except remove my previous version.
> 
> If I reboot kodi, my old version appears again.

It can take some minutes to install depending on your hardware unfortunately, not sure what the cause it. Whilst installing addon updates, kodi appears to uninstalls the previous version. If you wait long enough it should install.

**Comment by IANKA1007** [comment]
> > Surf to https://github.com/bbviking/plugin.video.seren Press the green Button "Code" Choose Download ZIP unzip that file to a new location rename plugin.video.seren-master to plugin.video.seren zip that again to plugin.video.seren.zip (note that the folder plugin.video.seren must be at the root of the zip) That zip file can now be used in kodi via install from zip so in fact you new created zip is the same as the downloaded one with the same structure except that plugin.video.seren-master is renamed to plugin.video.seren If you use 7-zip you can do that even without unzipping/zipping the downloaded zip
> 
> Wonderful!! Kudos on this one!! Install was flawless

This worked perfectly gor me too!!! Great instructions easy to follow.  I downloaded the zip on pc . Followed instructions , moved the new zip over to external HDD abd then plugged that into my Fire Cube and opens Kodi and lbrowsedd external HDD and unzipped and bam. All done!. Amazing.  Thank you kindly sir!

**Comment by liquidcms** [comment]
> > I have 3.0.5 installed. Add new source to bbviking's repo. When I add from zip and go to this source I see zip for 3.0.61. Selecting this does nothing except remove my previous version.
> > 
> > If I reboot kodi, my old version appears again.
> 
> It can take some minutes to install depending on your hardware unfortunately, not sure what the cause it. Whilst installing addon updates, kodi appears to uninstalls the previous version. If you wait long enough it should install.

Yes, sure enough, I just let it sit there and it did eventually finish. And works great. Thanks for the fix.

**Comment by bills176** [comment]
> It can take some minutes to install depending on your hardware unfortunately, not sure what the cause it. Whilst installing addon updates, kodi appears to uninstalls the previous version. If you wait long enough it should install.

My observation. I'm on a Win 11 box and the delay for installing seems to be downloading from the github source. After initiating I clicked back to my downloads and it displayed downloading xx% for a short time. Then I tried open a browser and navigate to the source github and click the zip to download and it is slow for a small file.

**Comment by Gillsiwillsi** [comment]
Just want to say a HUUUUUUGE thanks to bbviking for their fix. Worked a treat on all my Firesticks (including on Kodi 21). Awesome!! Takes 6-7 minutes.

**Comment by MoHara23** [comment]
Dear can I use the bbviking fix on my iPhone 11 Pro ? I use Kodi on this phone . Is there any step explanation how to add files to an iPhone and to make Kodi app see the file ?

**Comment by Gillsiwillsi** [comment]
> Dear can I use the bbviking fix on my iPhone 11 Pro ? I use Kodi on this phone . Is there any step explanation how to add files to an iPhone and to make Kodi app see the file ?

It worked great on my Android phone! 

Open Kodi

on the left hand side of the screen, tap on the Settings cog, then tap:

File Manager

Add source

<None>

and type in: 

https://bbviking.github.io/

then tap:

Enter a name 

Type: bb 

then tap: 

OK 

go back a screen, and then tap:

Addons 

Install from zip file 

bb

plugin.video.seren-3.0.62.zip

and wait...

It'll take a few minutes and I think you get a prompt to install/update Seren. Tap yes. 

I was worried I'd have to configure Seren all over again, but my accounts etc were intact.

**Comment by MoHara23** [comment]
> > Dear can I use the bbviking fix on my iPhone 11 Pro ? I use Kodi on this phone . Is there any step explanation how to add files to an iPhone and to make Kodi app see the file ?
> 
> It worked great on my Android phone!
> 
> Open Kodi
> 
> on the left hand side of the screen, tap on the Settings cog, then tap:
> 
> File Manager
> 
> Add source
> 
> and type in:
> 
> https://bbviking.github.io/
> 
> then tap:
> 
> Enter a name
> 
> Type: bb
> 
> then tap:
> 
> OK
> 
> go back a screen, and then tap:
> 
> Addons
> 
> Install from zip file
> 
> bb
> 
> plugin.video.seren-3.0.62.zip
> 
> and wait...
> 
> It'll take a few minutes and I think you get a prompt to install/update Seren. Tap yes.
> 
> I was worried I'd have to configure Seren all over again, but my accounts etc were intact.

Ok I did manage to install in anyway , I was a little confused : I thaught this fix is a workaround and needed manually installing files but it’s as easy as always : install from zip .

great solution .

only I  had to do extra : to install the Seren Omega add on .

thanks a lot

**Comment by xen0mk** [comment]
Can someone start to implement this in the main branch?

**Comment by moraisaugusto** [comment]
I can confirm that the zip file provided on https://bbviking.github.io worked fine!!! Took me a while to fix this issue. I updated my libreelec, changed my wireguard server. I was getting crazy. 😂

**Comment by Wlada** [comment]
It works for me as well. Thank you!

**Comment by xen0mk** [comment]
The zip worked only on 1 device out of 4 I tried... it just gets stuck on Installing....77% no matter how much time I leave it to go through it...

**Comment by Wlada** [comment]
It was the same for me, but just be patient. It takes some time.

**Comment by rattler98** [comment]
It takes a while to install, but it will finally get there if you're patient. Once you load it onto one device, use adblink to backup and clone onto your other devices.

**Comment by Jason-Onken** [comment]
Seems like we are getting issues again. Anyone else have problems. I have done the fix and started having issues yesterday when I tried a stream.

**Comment by balconor** [comment]
Mine stalls at 77% as well. Waiting overnight did not help.

**Comment by peno64** [comment]
It's a github thing.

Download the zip, put it on your kodi device and then install the zip.

**Comment by luizffgarcia** [comment]
This zip works well, its so fast finding playable sources its unbelievable. 

We should really get this fix merged into master, any reason not to?

**Comment by NorthEndSauce** [comment]
Is seren having issues again? I got the package and everything has worked great and now it will load tons of downloads and then it just won’t play anything I have been currently watching

**Comment by Mizokuiam** [comment]
Hi there @Arathonk! It looks like the "error code 37" and "no playable sources" issue in Seren is due to Real-Debrid disabling their `instantAvailability` API endpoint.  A temporary workaround is available, but official updates are recommended.

**Workaround (Use with caution):**

`henryjfry` provided a modified `real_debrid.py` file: [gist link](https://gist.github.com/henryjfry/725cce9ce0222b063e5e40a7c571d032)

1.  **Backup:** Before making any changes, back up your existing `real_debrid.py` file located in `~/.kodi/addons/plugin.video.seren/resources/lib/debrid/`.
2.  **Replace:** Replace the contents of your `real_debrid.py` with the content from the gist.
3.  **Timeout:**  You might need to increase your scraping timeout in Seren's settings.  The exact location of this setting may vary slightly depending on the Seren version.

**Important Considerations:**

*   This workaround might be based on an older Seren version.  Compatibility isn't guaranteed.
*   Modifying addon files can introduce instability. Proceed with caution and be prepared to restore your backup.
*   The best long-term solution is an official Seren update addressing these API changes. Keep an eye on the Seren repository for updates.
*   Real-Debrid's situation is evolving.  While torrent functionality might still exist, further changes could impact Seren and other addons.


If the workaround doesn't work or you're uncomfortable modifying addon files, waiting for an official Seren update is recommended.  In the meantime, alternative streaming solutions like Torrentio on Stremio might be viable options.


See if this works for you.

**Comment by PieterG75** [comment]
> Hi there [@Arathonk](https://github.com/Arathonk)! It looks like the "error code 37" and "no playable sources" issue in Seren is due to Real-Debrid disabling their `instantAvailability` API endpoint. A temporary workaround is available, but official updates are recommended.
> 
> **Workaround (Use with caution):**
> 
> `henryjfry` provided a modified `real_debrid.py` file: [gist link](https://gist.github.com/henryjfry/725cce9ce0222b063e5e40a7c571d032)
> 
> 1. **Backup:** Before making any changes, back up your existing `real_debrid.py` file located in `~/.kodi/addons/plugin.video.seren/resources/lib/debrid/`.
> 2. **Replace:** Replace the contents of your `real_debrid.py` with the content from the gist.
> 3. **Timeout:**  You might need to increase your scraping timeout in Seren's settings.  The exact location of this setting may vary slightly depending on the Seren version.
> 
> **Important Considerations:**
> 
> * This workaround might be based on an older Seren version.  Compatibility isn't guaranteed.
> * Modifying addon files can introduce instability. Proceed with caution and be prepared to restore your backup.
> * The best long-term solution is an official Seren update addressing these API changes. Keep an eye on the Seren repository for updates.
> * Real-Debrid's situation is evolving.  While torrent functionality might still exist, further changes could impact Seren and other addons.
> 
> If the workaround doesn't work or you're uncomfortable modifying addon files, waiting for an official Seren update is recommended. In the meantime, alternative streaming solutions like Torrentio on Stremio might be viable options.
> 
> See if this works for you.

Thanks for the heads up. The link to the file is dead though.

**Comment by fade2gray** [comment]
@PieterG75 https://github.com/nixgates/plugin.video.seren/issues/949#issuecomment-2502409119
